seq2seq/metrics/metrics.py

```diff
@@ -356,4 +356,297 @@ def eval_batch(self, outputs, targets):
+        output_eos_symbol (str): Output EOS symbol
+        output_unk_symbol (str): Output UNK symbol
+    """
+
+    _NAME = "Microtask Accuracy"
+    _SHORTNAME = "vp_tsk_acc"
+    _INPUT = "seqlist"
+
+    def __init__(self, input_vocab, output_vocab, use_output_eos, input_pad_symbol, output_sos_symbol, output_pad_symbol, output_eos_symbol, output_unk_symbol):
+        self.input_vocab = input_vocab
+        self.output_vocab = output_vocab
+
+        self.use_output_eos = use_output_eos
+
+        # instead of passing all these arguments, we could also hard-code to use <sos>, <pad>, <unk> and <eos>
+        self.input_pad_symbol = input_pad_symbol
+        self.output_sos_symbol = output_sos_symbol
+        self.output_pad_symbol = output_pad_symbol
+        self.output_eos_symbol = output_eos_symbol
+        self.output_unk_symbol = output_unk_symbol
+
+        self.seq_correct = 0
+        self.seq_total = 0
+
+        super(VerifyProduceAccuracy, self).__init__(self._NAME, self._SHORTNAME, self._INPUT)
+
+    def get_val(self):
+        """
+        Get the average accuracy metric of all processed batches
+        Returns:
+            float: average accuracy
+        """
+        if self.seq_total != 0:
+            return float(self.seq_correct) / self.seq_total
+        else:
+            return 0
+
+    def reset(self):
+        """
+        Reset after all batches have been processed
+        """
+        self.seq_correct = 0
+        self.seq_total = 0
+
+    def correct(self, grammar, prediction):
+        '''
+        Return True if the target is a valid output given the source
+        Args:
+            grammar (list(str)): List of symbols of the grammar
+            prediction (list(str)): List of symbols of the prediction
+        Returns:
+            bool: whether the prediction is coming from the grammar
+        '''
+        all_correct = False
+
+        any_in = lambda a, b: any(i in b for i in a)
+        all_in = lambda a, b: all(i in b for i in a)
+        not_in = lambda a,b: frozenset(a).isdisjoint(frozenset(b))
+        if ('or' in grammar):
+            target = list(set(grammar) - set(['produce', 'or']))
+            all_correct = any_in(prediction, target)
+        elif('and' in grammar and 'not' not in grammar):
+            target = list(set(grammar) - set(['produce', 'and']))
+            all_correct = all_in(prediction, target)
+        elif('not' in grammar):
+            grammar.pop(grammar.index('produce'))
+            target = grammar.copy() #list(set(grammar) - set(['produce']))
+            check1 = []
+            for i in range(1, len(target)):
+                if(target[i-1]=='not'):
+                    check1.append(target[i])
+            if(not_in(prediction, check1)):
+                ops = ['and', 'not']
+                ops.extend(check1)
+                check2 = list(set(target) - set(ops))
+                all_correct = True if all_in(check2, prediction) else False
```

Comment:
This line basically says `True if True else False`. You can remove the if-else

Reply:
fixed.